python/groupby.cpp

```diff
@@ -14,33 +14,75 @@ using namespace scipp::core;
 namespace py = pybind11;
 
 template <class T> void bind_groupby(py::module &m, const std::string &name) {
-  py::class_<GroupBy<T>> groupBy(m, name.c_str(), R"(
-    GroupBy object implementing to split-apply-combine mechanism.)");
-
-  groupBy.def("mean", &GroupBy<T>::mean);
-  groupBy.def("sum", &GroupBy<T>::sum);
-}
-
-void init_groupby(py::module &m) {
-  bind_groupby<DataArray>(m, "GroupByDataArray");
-  bind_groupby<Dataset>(m, "GroupByDataset");
-
   m.def("groupby",
-        py::overload_cast<const DatasetConstProxy &, const std::string &,
-                          const Dim>(&groupby),
-        py::arg("data"), py::arg("labels"), py::arg("combine"),
+        py::overload_cast<const typename T::const_view_type &,
+                          const std::string &, const Dim>(&groupby),
+        py::arg("data"), py::arg("group"), py::arg("combine"),
         py::call_guard<py::gil_scoped_release>(),
-        R"(Group dataset based on values of specified labels.
+        R"(
```

Comment:
It's a shame that the doc for an overloaded function (like the `args` and `kwargs`, and a long list of overloads...)
![Screenshot at 2019-11-05 13-43-58](https://user-images.githubusercontent.com/39047984/68208847-65f0b680-ffd2-11e9-9948-acf4d802a57c.png)



does not look as good as that of the functions below



![Screenshot at 2019-11-05 13-43-49](https://user-images.githubusercontent.com/39047984/68208873-73a63c00-ffd2-11e9-93df-e8370fc6edd0.png)

Reply:
I did try various things to make it look nicer, but failed. For some reason it refuses to make nice headings. Any ideas?

Reply:
no ideas :-(